### ISSUE TITLE
v1.2 Deployment e2e shouldn't run against against masters before v1.2

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#21421]
Right now, the following test is running from v1.2 clients against v1.1 masters:

**Kubectl client Kubectl run deployment should create a deployment from an image [Conformance]**

*/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:938 Expected error: <errors.errorString | 0xc20855cfd0>: { s: "Error running &{/jenkins-master-data/jobs/kubernetes-upgrade-gke-1.1-master-step1-deploy/workspace/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.136.3 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-upgrade-gke-1.1-master-step1-deploy/workspace/.kube/config run e2e-test-nginx-deployment --image=nginx --namespace=e2e-tests-kubectl-7chcz] [] Error from server: the server could not find the requested resource\n [] 0xc2085564a0 exit status 1 true [0xc20803ce60 0xc20803ce98 0xc20803cec8] [0xc20803ce60 0xc20803ce98 0xc20803cec8] [0xc20803ce80 0xc20803cec0] [0x886450 0x886450] 0xc208648420}:\nCommand stdout:\n\nstderr:\nError from server: the server could not find the requested resource\n\nerror:\nexit status 1\n", } Error running &{/jenkins-master-data/jobs/kubernetes-upgrade-gke-1.1-master-step1-deploy/workspace/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.136.3 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-upgrade-gke-1.1-master-step1-deploy/workspace/.kube/config run e2e-test-nginx-deployment --image=nginx --namespace=e2e-tests-kubectl-7chcz] [] Error from server: the server could not find the requested resource [] 0xc2085564a0 exit status 1 true [0xc20803ce60 0xc20803ce98 0xc20803cec8] [0xc20803ce60 0xc20803ce98 0xc20803cec8] [0xc20803ce80 0xc20803cec0] [0x886450 0x886450] 0xc208648420}: Command stdout: stderr: Error from server: the server could not find the requested resource error: exit status 1 not to have occurred*

My guess is that this test is not expected to pass against pre-v1.2 clusters.

Enabling deployments by default was incorporated in CI build `v1.2.0-alpha.7.726+0615f52054378a`.

cc @janetkuo

**Comment by janetkuo** [comment]
pre-v1.2 clusters don't have deployments enabled by default, so this test failed trying to create a deployment with `kubectl run`.

**Comment by bgrant0607** [comment]
@ihmccreery We're running the latest e2e tests on older clusters?

**Comment by ihmccreery** [comment]
We'd like to get some signal that latest `kubectl` works with older clusters.  e2e tests are the best we've got right now.  It's a hack for https://github.com/kubernetes/kubernetes/issues/16040.

**Comment by ihmccreery** [comment]
cc @jlowdermilk.

**Comment by ihmccreery** [comment]
Jeff & I were discussing this today.  Given the current tests, we're not convinced that a v1.2 client will be able to do a simple `kubectl run` on an older cluster and expect it to work.  That means when we release v1.2.

**Comment by bgrant0607** [comment]
Ref #22836

**Comment by bgrant0607** [comment]
@ihmccreery @jlowdermilk It's possible to bring back the older `kubectl run` test and pin to a generator.

We should be doing that, anyway: #20304

**Comment by bgrant0607** [comment]
We may want another test that tests that a pod runs rather than kubectl run produces a particular resource.

**Comment by jlowdermilk** [comment]
The backwards compatibility issue with v1.2 `kubectl run` against v1.1 master should be fixed by #22849. Ike, I think we can close this unless we want to add an e2e job that specifically tests 1.2 kubectl with 1.1 master. wdyt?

**Comment by jlowdermilk** [comment]
I just manually verified `kubectl run` works by default with 1.2 client and 1.1 master.

```
cluster/kubectl.sh version                
Client Version: version.Info{Major:"1", Minor:"2+", GitVersion:"v1.2.0-beta.1", GitCommit:"77eff081f022242e5736b8c2e1ca2058ca61a733", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.8", GitCommit:"a8af33dc07ee08defa2d503f81e7deea32dd1d3b", GitTreeState:"clean"}

cluster/kubectl.sh run nginx --image=nginx
replicationcontroller "nginx" created
```

I'm going to close this again, @ihmccreery and I can discuss offline whether/how to add e2e testing for client backwards compatibility.
